src/plugin.json

```diff
@@ -73,6 +78,224 @@
+      "grants": ["Admin", "Editor"]
+    },
+    {
+      "role": {
+        "name": "Alerts reader",
+        "description": "Read alerts in the Synthetic Monitoring app",
+        "permissions": [
+          { "action": "plugins.app:access", "scope": "plugins:id:grafana-synthetic-monitoring-app" },
+          { "action": "grafana-synthetic-monitoring-app:read" },
+
+          { "action": "grafana-synthetic-monitoring-app.alerts:read" }
+        ]
+      },
+      "grants": ["Viewer"]
+    },
+    {
+      "role": {
+        "name": "Alerts writer",
+        "description": "Create, edit and delete alerts in the Synthetic Monitoring app",
+        "permissions": [
+          { "action": "plugins.app:access", "scope": "plugins:id:grafana-synthetic-monitoring-app" },
+          { "action": "grafana-synthetic-monitoring-app:read" },
+          { "action": "grafana-synthetic-monitoring-app:write" },
+
+          { "action": "grafana-synthetic-monitoring-app.alerts:write" },
+          { "action": "grafana-synthetic-monitoring-app.alerts:read" },
+          { "action": "grafana-synthetic-monitoring-app.alerts:delete" }
+        ]
+      },
+      "grants": ["Admin", "Editor"]
+    },
+    {
+      "role": {
+        "name": "Thresholds reader",
+        "description": "Read thresholds in the Synthetic Monitoring app",
+        "permissions": [
+          { "action": "plugins.app:access", "scope": "plugins:id:grafana-synthetic-monitoring-app" },
+          { "action": "grafana-synthetic-monitoring-app:read" },
+
+          { "action": "grafana-synthetic-monitoring-app.thresholds:read" }
+        ]
+      },
+      "grants": ["Viewer"]
+    },
+    {
+      "role": {
+        "name": "Thresholds writer",
+        "description": "Read and edit thresholds in the Synthetic Monitoring app",
+        "permissions": [
+          { "action": "plugins.app:access", "scope": "plugins:id:grafana-synthetic-monitoring-app" },
+          { "action": "grafana-synthetic-monitoring-app:read" },
+          { "action": "grafana-synthetic-monitoring-app:write" },
+
+          { "action": "grafana-synthetic-monitoring-app.thresholds:write" },
+          { "action": "grafana-synthetic-monitoring-app.thresholds:read" }
+        ]
+      },
+      "grants": ["Admin", "Editor"]
+    },
+    {
+      "role": {
+        "name": "Access tokens writer",
+        "description": "Create and delete access tokens in the Synthetic Monitoring app",
+        "permissions": [
+          { "action": "plugins.app:access", "scope": "plugins:id:grafana-synthetic-monitoring-app" },
+          { "action": "grafana-synthetic-monitoring-app:read" },
+          { "action": "grafana-synthetic-monitoring-app:write" },
+
+          { "action": "grafana-synthetic-monitoring-app.access-tokens:write" },
+          { "action": "grafana-synthetic-monitoring-app.access-tokens:delete" }
+        ]
+      },
+      "grants": ["Admin"]
+    },
+    {
+      "role": {
+        "name": "Access tokens reader",
```

Comment:
Where can access tokens be read? iirc they are only visible when they are generated. If they can be read as an editor, does that essentially mean the user can escalate themself to admin by using the token with the API and avoid RBAC?

Reply:
They can't be read yet but they should be able to. I probably have dozens (hundreds...) of tokens in my account which I can't revoke and I would love to see them so I can delete them all.

I do find this a difficult part to get my head round RBAC at times. Does it make sense to have separate roles for reading / writing / deleting access tokens or should this be a simple switch of you either have complete control to access tokens or you don't?

I think I can see a use case for saying "I'll grant you access to read our existing tokens so you can set up our SM instance but I don't want you to create any more." and that follows the conventions for all the other RBAC roles.

Reply:
I have removed the Access Token Reader role as there is no use for it right now and it might confuse users to see it in the list of available roles to assign. We can add it back once we implement the view to actually list and delete them.

Reply:
Yup, makes sense. It’s easy to add roles later